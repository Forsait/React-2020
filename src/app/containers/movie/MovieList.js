// @flow
import React from 'react';

import styles from './MovieList.module.scss';
import Movie from '../../components/movieItem/Movie';

import type { Movie as MovieType } from '../../models/movie';

type Props = {
  movieArr: MovieType[]
}

export default class MovieList extends React.Component<Props> {
  constructor(props: Props) {
    super(props);
  }

  render() {
    let moviesResult;
    if (this.props.movieArr.length > 0) {
      const movies = this.props.movieArr.map((item) => <Movie key={item.id} data={item} />);
      moviesResult = (
        <div className={styles.movie_container}>
          {movies}
        </div>
      );
    } else {
      moviesResult = (
        <div className={styles.not_found}>
          No films found
        </div>
      );
    }

    return (
      <div className={styles.movies}>
        {moviesResult}
      </div>
    );
  }
}
