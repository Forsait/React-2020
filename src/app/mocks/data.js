export const film = {
  budget: 0,
  genres: ['Action', 'Adventure', 'Drama', 'Science Fiction', 'Thriller'],
  id: 351286,
  overview: 'A volcanic eruption threatens',
  poster_path: 'https://image.tmdb.org/t/p/w500/ln6d5Okr6VK5vfQVobJTiYxeD0l.jpg',
  release_date: '2018-06-01',
  revenue: 0,
  runtime: null,
  tagline: 'Life finds a way',
  title: 'Jurassic World: Fallen Kingdom',
  vote_average: 0,
  vote_count: 28,
};

export const moviesData = {
  data: new Array(6).fill(film),
  limit: 0,
  offset: 0,
  total: 0,
};

export const radioOptions = {
  title: 'Radio title',
  list: [
    {
      key: 'key 1',
      title: 'title 1',
    },
    {
      key: 'key 2',
      title: 'title 2',
    },
  ],
};
