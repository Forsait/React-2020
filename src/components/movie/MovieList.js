import React from 'react';

import styles from './MovieList.module.scss';
import Movie from './Movie';

export default class MovieList extends React.Component{

  constructor(props){
    super(props);
  }

  render() {
    let moviesResult;
    if (this.props.movieArr.length > 0) {
      const movies = this.props.movieArr.map((item) => {
        return <Movie key={item.id} data={item} />
      });
      moviesResult = <div className={styles.movie_container}>
        {movies}
      </div>
    } else {
      moviesResult = <div className={styles.not_found}>
        No films found
      </div>
    }

    return <div className={styles.movies}>
      {moviesResult}
    </div>
  }
}