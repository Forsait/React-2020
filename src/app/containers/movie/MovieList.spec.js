import React from 'react';
import { shallow } from 'enzyme';

import MovieList from './MovieList';
import { film } from '../../mocks/data';

describe('MovieList component', () => {
  it('should be render correctly', () => {
    const com = shallow(<MovieList movieArr={new Array(6).fill(film)} />);
    expect(com).toMatchSnapshot();
  });

  it('should be render correctly with no films', () => {
    const com = shallow(<MovieList movieArr={[]} />);
    expect(com).toMatchSnapshot();
  });
});
