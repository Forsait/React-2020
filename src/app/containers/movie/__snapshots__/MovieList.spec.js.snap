// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList component should be render correctly 1`] = `
<div>
  <div>
    No films found
  </div>
</div>
`;
